framework/src/controller/application.js

```diff
@@ -232,26 +225,32 @@ class Application {
 	 * @param {number} transactionType - Unique integer that identifies the transaction type
 	 * @param {constructor} Transaction - Implementation of @liskhq/lisk-transactions/base_transaction
 	 */
-	registerTransaction(transactionType, Transaction, options = {}) {
-		// TODO: Validate the transaction is properly inherited from base class
+	registerTransaction(Transaction, matcher) {
+		assert(Transaction, 'Transaction implementation is required');
+
+		assert(
+			Transaction.prototype instanceof BaseTransaction,
+			'Transaction must extend BaseTransaction.'
+		);
+
 		assert(
-			Number.isInteger(transactionType),
+			Number.isInteger(Transaction.TYPE),
 			'Transaction type is required as an integer'
 		);
+
 		assert(
-			!Object.keys(this.getTransactions()).includes(transactionType.toString()),
-			`A transaction type "${transactionType}" is already registered.`
+			!Object.keys(this.getTransactions()).includes(Transaction.TYPE),
+			`A transaction type "${Transaction.TYPE}" is already registered.`
 		);
-		assert(Transaction, 'Transaction implementation is required');
 
-		if (options.matcher) {
+		if (matcher) {
 			Object.defineProperty(Transaction.prototype, 'matcher', {
```

Comment:
This line refers to the fact that `matcher` can be defined in the `Transaction` class itself, instead of specifying it at time of registration.  Yes when you define at time of registration, it must override actual getter if exists. 

@MaciejBaj We should document it properly.